docker/mnt/run.sh

```diff
@@ -0,0 +1,121 @@
+#! /bin/bash
+set -e
+
+if [ -z "$1" ]; then
+    # the main container need not stay running for services
+    exit 0
+fi
+
+function setup() {
+    [ -n "$1" ] && TEST="$1"
+
+    if [[ "$TEST" =~ ^python ]]; then
+
+        scripts/uninstall-requirements.sh
+        pip install \
+            -r requirements/requirements.txt \
+            -r requirements/dev-requirements.txt \
+            coveralls
```

Comment:
Yes, since the base image is built frequently the python env should be basically up to date (except for any changes in the current branch). I don't think caching the virtual env will make that much difference since you'll still need to run this same command on every build.